### PR TITLE
Redesign front page

### PR DESCRIPTION
**Pull request by randombenj**
Redesign the front page to make it more searchable:

- Favourites are listed by default
- Search for projects
- Search queries can be shared
- Improve UI

**TODO:**
- [x] Contact information
- [x] Show all docs after Favourites
- [x] @fliiiix review size implementation
- [x] Investigate centering
- [x] Upload/Claim/... -> 404
- [x] Docs button -> Hover and less extreme border
- [x] Add basic auth example to README.md
- [x] Update tesaser GIF **before merging**
- [x] @fliiiix  fix /api/stats cache invalidation

docat

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,9 +5,22 @@
 [![build](https://github.com/docat-org/docat/workflows/docat%20ci/badge.svg)](https://github.com/docat-org/docat/actions)
 [![Gitter](https://badges.gitter.im/docat-docs-hosting/community.svg)](https://gitter.im/docat-docs-hosting/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
 
+## Why DoCat?
```
Comment (timofurrer) [inline README.md:8]: Is `DoCat` really the casing? Haven't seen it anywhere else ...

File: README.md [hunk 2]
```diff
@@ -5,9 +5,22 @@
...
+
+When generating static developer documentation using
```
Comment (timofurrer) [inline README.md:10]:
```suggestion
When generating static documentation using
```

File: README.md [hunk 3]
```diff
@@ -5,9 +5,22 @@
...
+[mkdocs](https://www.mkdocs.org/), [sphinx](http://www.sphinx-doc.org/en/master/), ...
+hosting just one version of the docs might not be enough.
+Many users might still use older versions and might need to read
+those versions of the documentation.
+
+Docat solves this problem by providing a simple tool that can
```
Comment (timofurrer) [inline README.md:16]:
```suggestion
Docat solves this problem by providing a simple tool that
```

File: README.md [hunk 4]
```diff
@@ -5,9 +5,22 @@
...
+host multiple documentation projects with multiple versions.
```
Comment (timofurrer) [inline README.md:17]:
```suggestion
hosts multiple documentation projects with multiple versions.
```

File: README.md [hunk 5]
```diff
@@ -5,9 +5,22 @@
...
+
+*The main design decision with docat was to keep the tool as simpel as possible*
```
Comment (timofurrer) [inline README.md:19]:
```suggestion
*The main design decision with docat was to keep the tool as simple as possible.*
```

File: README.md [hunk 6]
```diff
@@ -23,75 +36,43 @@ docker run \
...
 
-### Local Development
+### Using DoCat
 
-For local development, first configure and start the backend (inside the `docat/` folder):
-
-```sh
-# create a folder for local development (uploading docs)
-DEV_DOCAT_PATH="$(mktemp -d)"
+> 🛈 Please note that docat does not provide any way to write documentation
```
Comment (timofurrer) [inline README.md:43]:
```suggestion
> 🛈 Please note that docat does not provide any way to write documentation.
```

File: README.md [hunk 7]
```diff
@@ -23,75 +36,43 @@ docker run \
...
+> the tool only hosts documentation.
```
Comment (timofurrer) [inline README.md:44]:
```suggestion
> It's sole responsibility is to host documentation.
```

File: README.md [hunk 8]
```diff
@@ -23,75 +36,43 @@ docker run \
...
+>
+> There are many awesome tools to write developer documenation:
```
Comment (timofurrer) [inline README.md:46]:
```suggestion
> There are many awesome tools to write documenation:
```

File: README.md [hunk 9]
```diff
@@ -23,75 +36,43 @@ docker run \
...
+> [mkdocs](https://www.mkdocs.org/), [sphinx](http://www.sphinx-doc.org/en/master/),
+> [mdbook](https://rust-lang.github.io/mdBook/) ...
```
Comment (timofurrer) [inline README.md:48]:
```suggestion
> - [mkdocs](https://www.mkdocs.org/)
> - [sphinx](http://www.sphinx-doc.org/en/master/)
> - [mdbook](https://rust-lang.github.io/mdBook/)
> - ...
```

File: README.md [hunk 10]
```diff
@@ -23,75 +36,43 @@ docker run \
...
 
-# install dependencies
-poetry install
 
-# run the local development version
-DOCAT_SERVE_FILES=1 DOCAT_STORAGE_PATH="$DEV_DOCAT_PATH" poetry run python -m docat
-```
+A small tool called [docatl](https://github.com/docat-org/docatl) is provided
```
Comment (timofurrer) [inline README.md:51]:
```suggestion
A CLI tool called [docatl](https://github.com/docat-org/docatl) is available
```

File: README.md [hunk 11]
```diff
@@ -23,75 +36,43 @@ docker run \
...
+for easy interaction with the docat server.
+However, interacting with docat can also be done through [`curl`](doc/getting-started.md).
 
-After this you need to start the frontend (inside the `web/` folder):
+So in order to push documentation (and tag as `latest`) in the folder `docs/` simply run:
```
Comment (timofurrer) [inline README.md:55]:
```suggestion
To push documentation (and tag as `latest`) in the folder `docs/` simply run:
```

File: web/src/components/Project.tsx [hunk 12]
```diff
@@ -1,64 +1,88 @@
...
-import FavoriteStar from './FavoriteStar'
 import { Tooltip } from '@mui/material'
+import FavoriteStar from './FavoriteStar'
 
 interface Props {
   project: ProjectType
   onFavoriteChanged: () => void
 }
 
+function timeSince(date: Date) {
```
Comment (fliiiix) [inline web/src/components/Project.tsx:14]: js is crazy

File: README.md [hunk 13]
```diff
@@ -36,7 +36,7 @@ docker run \
 
 Go to [localhost:8000](http://localhost:8000) to view your docat instance:
 
-![docat screenshot](doc/assets/docat-screenshot.png)
+<img src="doc/assets/docat.gif" width="100%" />
```
Comment (fliiiix) [inline README.md:39]: love it

File: docat/docat/utils.py [hunk 14]
```diff
@@ -125,6 +125,54 @@ def is_forbidden_project_name(name: str) -> bool:
...
+        size_suffix = " bytes"
+
+    return str(amount) + size_suffix
+
+
+def directory_size(path: Path) -> int:
+    return sum(file.stat().st_size for file in path.rglob("*") if file.is_file())
+
+
+def get_system_stats(upload_folder_path: Path) -> Stats:
```
Comment (fliiiix) [inline docat/docat/utils.py:179]: todo felix

Comment (fliiiix) [reply docat/docat/utils.py:179]: NotADirectoryError: [Errno 20] Not a directory: 'doc/config.json'

**Comment by fliiiix** [comment]
image
 
only having hashtags sometimes is on purpose right?

File: README.md [hunk 15]
```diff
@@ -23,75 +36,102 @@ docker run \
...
+  ```
 
-## Advanced Frontend `config.json`
+3) Mounted to the correct location inside the container:
+
+  ```
+  docker run \
+    --detach \
+    --volume $PWD/docat-run:/var/docat/ \
+    --volume $PWD/nginx/default.conf:/etc/nginx/http.d/default.conf \
```
Comment (fliiiix) [inline README.md:125]:
```suggestion
    --volume $PWD/nginx/default.conf:/app/docat/docat/nginx/default \
```

since the env substitution

**Comment by fliiiix** [comment]
Latest state deployed to test and @randombenj i reported a long project name which breaks the layout a bit up to you if you want to fix or not

**Pull request merged**